### PR TITLE
:book: Starting best practices

### LINKED ISSUE #2006: OCM ManifestWork resources size limit
As seen in a log file shared by @clubanderson:

> E0402 13:39:29.737793       1 generic_transport_controller.go:248] "failed to process Binding object" err="failed to process object with key 'kuberay-bindingpolicy' - failed to propagate wrapped object 'kuberay-bindingpolicy-wds1' to all required WECs - failed to propagate wrapped object to cluster mailbox namespace 'edgeplatform-test-wec1' - failed to create wrapped object 'kuberay-bindingpolicy-wds1' in destination WEC mailbox namespace 'edgeplatform-test-wec1' - admission webhook \"manifestworkvalidators.admission.work.open-cluster-management.io\" denied the request: the size of manifests is 1042259 bytes which exceeds the 512000 limit" logger="transport-controller" objectKey="kuberay-bindingpolicy" workerID=4

We can see that the cap on manifests size is `512000 bytes`, which is smaller than the kubernetes resource size cap (`1MiB`).

### PR DESCRIPTION
**Pull request by ezrasilvera**
## Related issue(s)

Fixes #2006

### CODE REVIEW

File: docs/content/direct/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Best Practices
+
+**Note:**  This section is under constration and includes partial information. 
+## Size considerations
+As KubeStellar is built on top of Kubernetes all Kuberentes limitations and recommendations apply to KubeStellar as well. These recommendations can be found in [Kubernetes Considerations for large clusters](https://kubernetes.io/docs/setup/best-practices/cluster-large/).
+
+In addition, KubeStellar Transport Plugin is built on top of OCM, so KubeStellar also comply to some of OCM's limitations. Users should take into account the following restrictions:
```
Comment (effi-ofer) [inline docs/content/direct/best-practices.md:7]: Link to OCM's limitations?

Comment (ezrasilvera) [reply docs/content/direct/best-practices.md:7]: Didn't find one :(

Comment (effi-ofer) [reply docs/content/direct/best-practices.md:7]: I couldn't find the limits documentation either, but note that in the ocm code, the limit is 500K not 512K https://github.com/open-cluster-management-io/ocm/blob/ff77abd31208aaf328b1dc0826b66caf9cdf6dcd/pkg/work/webhook/common/validator.go

**Review by effi-ofer — APPROVED** [review]

**Pull request merged**